aep/general/0134/aep.md.j2

```diff
@@ -0,0 +1,321 @@
+}
+```
+
+{% tab oas %}
+
+**Note:** OAS example not yet written.
+
+{% endtabs %}
+
+More specifically, the `allow_missing` flag triggers the following behavior:
+
+- If the method call is on a resource that does not exist, the resource is
+  created. All fields are applied regardless of any provided field mask.
+  - However, if any required fields are missing or fields have invalid values,
+    an `INVALID_ARGUMENT` error is returned.
+- If the method call is on a resource that already exists, and all fields
+  match, the existing resource is returned unchanged.
+- If the method call is on a resource that already exists, only fields declared
+  in the field mask are updated.
+
+The user **must** have the update permissions to call `Update` even with
+`allow_missing` set to `true`.
+
+### Etags
+
+An API may sometimes need to allow users to send update requests which are
+guaranteed to be made against the most current data (a common use case for this
+is to detect and avoid race conditions). Resources which need to enable this do
+so by including a `string etag` field, which contains an opaque,
+server-computed value representing the content of the resource.
+
+In this situation, the resource **should** contain a `string etag` field:
+
+{% tab proto %}
+
+```proto
+message Book {
+  option (google.api.resource) = {
+    type: "library.example.com/Book"
+    pattern: "publishers/{publisher}/books/{book}"
+  };
+
+  // The resource path of the book.
+  // Format: publishers/{publisher}/books/{book}
+  string path = 1 [(google.api.field_behavior) = IDENTIFIER];
+
+  // The title of the book.
+  // Example: "Mary Poppins"
+  string title = 2;
+
+  // The author of the book.
+  // Example: "P.L. Travers"
+  string author = 3;
+
+  // The etag for this book.
+  // If this is provided on update, it must match the server's etag.
+  string etag = 4;
+}
+```
+
+{% tab oas %}
+
+**Note:** OAS example not yet written.
+
+{% endtabs %}
+
+The `etag` field **may** be either [required][] or [optional][]. If it is set,
+then the request **must** succeed if and only if the provided etag matches the
+server-computed value, and **must** fail with an `ABORTED` error otherwise. The
+`update_mask` field in the request does not affect the behavior of the `etag`
+field, as it is not a field _being_ updated.
+
+### Expensive fields
+
+APIs sometimes encounter situations where some fields on a resource are
+expensive or impossible to reliably return.
+
+This can happen in a few situations:
+
+- A resource may have some fields that are very expensive to compute, and that
+  are generally not useful to the customer on update requests.
+- A single resource sometimes represents an amalgamation of data from multiple
+  underlying (and eventually consistent) data sources. In these situations, it
+  may be infeasible to return authoritative information on the fields that were
+  not changed.
+
+In this situation, an API **may** return back only the fields that were updated
```

Comment:
this guidance is pretty declarative-unfriendly, and I think we should revisit this.

user-settable values shouldn't need to be "computed" - that's what state values are for.

But I'd imagine this is a larger conversation, so file a ticket?

Reply:
Yeah an issue would be great.  Not filing one because I'm not sure I exactly understand which guidance you're referring to (even though I agree that "user-settable values shouldn't need to be 'computed'").

Reply:
Sorry, I'm referring specifically to the guidance that update can return partial resources.  Yes I'll file an issue on this.